spotinst/resource_spotinst_ocean_right_sizing_rule.go

```diff
@@ -521,9 +723,41 @@ func expandLabelsForDetach(data interface{}) ([]*right_sizing.Label, error) {
 		if v, ok := attr[string(ocean_right_sizing_rule.ValueForDetach)].(string); ok && v != "" {
 			label.Value = spotinst.String(v)
 		}
-
-		labels = append(labels, label)
-
+		//In case of responseDataLabel empty list not going to validate user's workload again GET attachedWorkload response in IF block.
```

Comment:
correct the statement as mentioned in attach workload

Reply:
updated to : //In case of responseDataLabel returned as empty list then not validating user's workload against GET attachedWorkload response.